auth/src/main/java/org/openedx/auth/presentation/signin/compose/SignInView.kt

```diff
@@ -0,0 +1,403 @@
+                        )
+                        Spacer(modifier = Modifier.height(24.dp))
+                        AuthForm(
+                            buttonWidth,
+                            state,
+                            onEvent,
+                        )
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun AuthForm(
+    buttonWidth: Modifier,
+    state: SignInUIState,
+    onEvent: (AuthEvent) -> Unit,
+) {
+    var login by rememberSaveable { mutableStateOf("") }
+    var password by rememberSaveable { mutableStateOf("") }
+
+    Column(horizontalAlignment = Alignment.CenterHorizontally) {
+        LoginTextField(
+            modifier = Modifier
+                .fillMaxWidth(),
+            onValueChanged = {
+                login = it
+            })
+
+        Spacer(modifier = Modifier.height(18.dp))
+        PasswordTextField(
+            modifier = Modifier
+                .fillMaxWidth(),
+            onValueChanged = {
+                password = it
+            },
+            onPressDone = {
+                onEvent(AuthEvent.SignIn(login = login, password = password))
+            }
+        )
+
+        Row(
+            Modifier
+                .fillMaxWidth()
+                .padding(top = 20.dp, bottom = 36.dp)
+        ) {
+            Text(
+                modifier = Modifier.noRippleClickable {
+                    onEvent(AuthEvent.RegisterClick)
+                },
+                text = stringResource(id = R.string.auth_register),
+                color = MaterialTheme.appColors.primary,
+                style = MaterialTheme.appTypography.labelLarge
+            )
+            Spacer(modifier = Modifier.weight(1f))
+            Text(
+                modifier = Modifier.noRippleClickable {
+                    onEvent(AuthEvent.ForgotPasswordClick)
+                },
+                text = stringResource(id = R.string.auth_forgot_password),
+                color = MaterialTheme.appColors.primary,
+                style = MaterialTheme.appTypography.labelLarge
+            )
+        }
+
+        if (state.showProgress) {
+            CircularProgressIndicator(color = MaterialTheme.appColors.primary)
+        } else {
+            OpenEdXButton(
+                width = buttonWidth,
+                text = stringResource(id = R.string.auth_sign_in),
+                onClick = {
+                    onEvent(AuthEvent.SignIn(login = login, password = password))
+                }
+            )
+        }
+        if (state.shouldShowSocialLogin) {
+            SocialLoginView(buttonWidth = buttonWidth, onEvent = onEvent)
+        }
+    }
+}
+
+@Composable
+private fun SocialLoginView(
+    buttonWidth: Modifier,
+    onEvent: (AuthEvent) -> Unit,
+) {
+    OpenEdXOutlinedButton(
+        modifier = buttonWidth.padding(top = 24.dp),
+        backgroundColor = colorResource(id = org.openedx.core.R.color.background),
```

Comment:
please use colors from the theme or defined in Colors.kt instead from xml.

Reply:
done